### PR TITLE
Issue 112 jackknife

### LINKED ISSUE #112: Jackknife estimate of the variance
I do not get the same estimate of the sd using the jackknife option of PLN and doing it by hand. Did I do smthing wrong?

```r
# Test Jackknife variance estimate

library(PLNmodels)

# Make parms
data("barents")
n <- nrow(barents$Abundance); p <- ncol(barents$Abundance)
fit <- PLN(barents$Abundance ~ barents$Latitude + barents$Longitude + barents$Depth + barents$Temperature)
X <- lm(barents$Abundance[, 1] ~ barents$Latitude + barents$Longitude + barents$Depth + barents$Temperature, x=TRUE)$x

# Simulate data
Y <- matrix(rpois(n*p, exp(X%*%fit$model_par$B + mvtnorm::rmvnorm(n, sigma=fit$model_par$Sigma))), n, p)
d <- ncol(X)

# Jacknife sd estimate
pln <- PLN(Y ~ -1 + X, control=PLN_param(config_post = list(jackknife=TRUE)))
betaSd1 <- as.vector(standard_error(pln, type='jackknife'))

# Jacknife sd estimate by hand
betaJK <- matrix(NA, n, p*d)
for(i in 1:n){
  plnTmp <- PLN(Y[-i, ] ~ -1 + X[-i, ])
  betaJK[i, ]  <- as.vector(plnTmp$model_par$B)
}
betaSd2 <- sqrt((n-1)*(colMeans(betaJK^2) - (colMeans(betaJK)^2)))

# Comparison
plot(betaSd1, betaSd2)
```

### PR DESCRIPTION
**Pull request by mahendra-mariadassou**
This PR fixes #112 by 
- extracting the computations for the starting point of $B$, $M$ and $S$ and into their own function
- using it in resampling-based variance estimators

Jackknife and bootstrap variance / sd estimators returned with `standard_error(myPLN, "bootstrap")` or `standard_error(myPLN, "jackknife")` are now identical to the ones computed "manally" with repeated call to `PLN()`

For @jchiquet, you can ignore all commits made before [df51164f5c696cc7262ba9b8c74b40b5e66e5981] / 17th of November, they have been reverted back in further commits.

### CODE REVIEW

File: R/PLNfit-class.R [hunk 1]
```diff
@@ -205,13 +209,17 @@ PLNfit <- R6Class(
     },
 
     variance_jackknife = function(Y, X, O, w, config = config_default_nlopt) {
+      cat("\n  Computing jackknife variance estimator", sep = "\n")
```
Comment (jchiquet) [inline R/PLNfit-class.R:212]: Could you add a condition to display the message (based on the config argument) ??

**Review by jchiquet — COMMENTED** [review]
minor comment on when/how displaying messages

File: R/PLNfit-class.R (hunk 1 above)
Comment (mahendra-mariadassou) [reply R/PLNfit-class.R:212]: Argh, this a leftover from my debugging session. I'm inclined to remove it completely. Do you think we should keep it (if so, I'll add the same for bootstrap) ?

**Review by jchiquet — COMMENTED** [review]
Thanks, you should also add an entry into the _pkgdown.yml file to avoid the corresponding action to fail.

File: R/PLNfit-class.R (hunk 1 above)
Comment (jchiquet) [reply R/PLNfit-class.R:212]: removing is perfectly fine.

**Review by jchiquet — APPROVED** [review]

**Review by jchiquet — APPROVED** [review]

**Pull request merged**